src/Microsoft.AspNetCore.Routing/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -250,15 +250,13 @@ public PolicyJumpTable BuildJumpTable(int exitDestination, IReadOnlyList<PolicyJ
         private Endpoint CreateRejectionEndpoint(IEnumerable<string> httpMethods)
         {
             var allow = string.Join(", ", httpMethods);
-            return new RouteEndpoint(
+            return new Endpoint(
```

Comment:
This provides test coverage. Think we need more specific non-RouteEndpoint tests?

Reply:
I think it would be good to have one unit test in case this implementation changes more. Curently the only way to do this is with a policy - see TestMetadata1MatcherPolicy in DfaMatcherBuilderTest